include/arch/arm/arch.h

```diff
@@ -43,9 +43,11 @@ extern "C" {
 #include <arch/arm/cortex_m/nmi.h>
 #endif
 #if defined(CONFIG_MPU_STACK_GUARD)
-#define STACK_ALIGN  32
+#define STACK_ALIGN		32
+#define STACK_GUARD_SIZE	4
```

Comment:
The stack we are defining is some user defined length.  The minimum alignment for any MPU region is 32 bytes, regardless of the unique capabilities of the MPU.  The minimum region size is 32 for ARM MPUs but not for one of the NXP ones

So you're right we have to account for this.